tests/integrations/android_sdk/test_verify_android_sdk.py

```diff
@@ -0,0 +1,281 @@
+    command.sys = MagicMock()
+
+    # Use the `tmp_path` in `dot_briefcase_path` to ensure tests don't interfere
+    # with each other.
+    command.dot_briefcase_path = tmp_path / ".briefcase"
+
+    # Make the `os` module and `host_os` live.
+    command.os = os
+
+    # Mock a host platform
+    command.host_os = 'Unknown'
+
+    # Override some other modules so we can test side-effects.
+    command.download_url = MagicMock()
+    command.subprocess = MagicMock()
+    command.shutil = MagicMock()
+
+    return command
+
+
+def accept_license(root_path):
+    "Generate a side effect method that will accept a license."
+    def _side_effect(*args, **kwargs):
+        license_dir = root_path / "licenses"
+        license_dir.mkdir(parents=True)
+        (license_dir / "android-sdk-license").touch()
+
+    return _side_effect
+
+
+@pytest.mark.parametrize("host_os", ("ArbitraryNotWindows", "Windows"))
+def test_succeeds_immediately_in_happy_path(mock_command, host_os, tmp_path):
+    "If verify_android_sdk is invoked on a valid path, it does nothing."
+    # `sdkmanager` exists and has the right permissions, and
+    # `android-sdk-license` exists, verify_android_sdk() should
+    # succeed, create no subprocesses, make no requests, and return a
+    # SDK wrapper.
+
+    # On Windows, this requires `sdkmanager.bat`; on non-Windows, it requires
+    # `sdkmanager`.
+
+    # Create `sdkmanager` and the license file.
+    root_path = tmp_path / ".briefcase" / "tools" / "android_sdk"
+    tools_bin = root_path / "tools" / "bin"
+    tools_bin.mkdir(parents=True, mode=0o755)
+    if host_os == "Windows":
+        (tools_bin / "sdkmanager.bat").touch()
+    else:
+        (tools_bin / "sdkmanager").touch(mode=0o755)
+    licenses = root_path / "licenses"
+    licenses.mkdir(parents=True, mode=0o755)
+    (licenses / "android-sdk-license").touch()
+
+    # Configure `mock_command` to assume the `host_os` we parameterized with.
+    mock_command.host_os = host_os
+
+    # Expect verify_android_sdk() to succeed
+    sdk = verify_android_sdk(mock_command)
+
+    # No calls to download, run or unpack anything.
+    mock_command.download_url.assert_not_called()
+    mock_command.subprocess.run.assert_not_called()
+    mock_command.subprocess.check_output.assert_not_called()
+    mock_command.shutil.unpack_archive.assert_not_called()
+
+    # The returned SDK has the expected root path.
+    assert sdk.root_path == root_path
+
+
+def test_user_provided_sdk(mock_command, tmp_path):
+    "If the user specifies a valid ANDROID_SDK_ROOT, it is used"
+    # Create `sdkmanager` and the license file.
+    root_path = tmp_path / "other_sdk"
+    tools_bin = root_path / "tools" / "bin"
+    tools_bin.mkdir(parents=True, mode=0o755)
+    (tools_bin / "sdkmanager").touch(mode=0o755)
+    licenses = root_path / "licenses"
+    licenses.mkdir(parents=True, mode=0o755)
+    (licenses / "android-sdk-license").touch()
```

Comment:
I'd prefer you mock-out the license check; otherwise, this is a lot of boilerplate.

Reply:
It's a little difficult to mock, as we're in the process of creating the object that will have the license check invoked on it; but I *can* use the pre-existing side effect method that sets up the signed license.